cypress/tests/module_test/runner.js

```diff
@@ -1,9 +1,26 @@
 const cypress = require('cypress')
 
-cypress.run({
-  headless: true,
-}).then(result => {
-  if (result.status === 'failed') {
-    process.exit(1);
+async function main() {
+  try {
+      const result = await cypress.run({
+          headless: true,
+      })
+
+      // If any tests have failed, results.failures is non-zero, some tests have failed
+      if (result.failures) {
+          console.error('One or more cypress tests have failed')
+          console.error(result.message)
+          process.exit(1)
+      }
+
+      if (result.status !== 'finished') {
+          console.error('Cypress tests failed with status', result.status)
+          process.exit(2)
+      }
+  } catch (e) {
+      console.error("Cypress encountered unexpected exception. Exiting.", e)
+      process.exit(3)
   }
-})
\ No newline at end of file
+}
+
+main();
```

Comment:
Might be cleaner & more robust to not assume that any exceptions in main are caught and do this instead.

```
main().catch(e => {
      console.error("Cypress encountered unexpected exception. Exiting.", e)
      process.exit(3)
});
```

and then you can remove the try/catch block from main itself

Reply:
Or to make main() even cleaner and to move to responsibility for setting the process exitCode outside of main you could do this,

```
main()
    .then(code => process.exitCode = code)
    .catch(e => {
        console.error("Cypress encountered unexpected exception. Exiting.", e)
        process.exitCode = 3
    });
```

and then `main()` returns the exitCode instead of calling `process.exit()`